### PR TITLE
Add interface sorting support.

### PR DESCRIPTION
**Pull request by dgjustice**
@qduk I have refactored the gist I sent (thank you for pointing out how broken that was! :grimacing:).  I finally put together a proper PR with all the test cases I could find.  This is a bit more complexity, but taking the "character class" approach allows future modifications to completely customize the sorting logic based on arbitrary groups of characters.  Sorts happen first by `CharacterClass.weight`, and secondly by whatever logic you like, for example, casting to `int` for the `CCInt` class.

### CODE REVIEW

**Review by qduk — CHANGES_REQUESTED** [review]
@dgjustice gotta say, this is pretty awesome stuff. I've just got one other note. 

Docs is picking up on your character classes and helper functions. I personally don't think those need to be in there. I am able to get them undocumented by making them private functions. I'll defer to @itdependsnetworks and @jeffkala to see if they agree or not. 

Thanks again @dgjustice. Once this is merged it'll allow me to build out my PR.

**Comment by dgjustice** [comment]
Travis is being flaky again.  I think the last commit is clean (passes tests); the build failed on installing dependencies.

**Review by qduk — APPROVED** [review]
@jeffkala You make a good point. Never hurts to have the extra documentation. 
@dgjustice Thanks for adding the docstring example.

File: netutils/interface.py
Comment (itdependsnetworks) [inline netutils/interface.py:269]: Can we mark split_interface_tuple and insert_nodes as private? I rather be explicit that these are not supported functions and subject to change.

Comment (dgjustice) [reply netutils/interface.py:269]: Yes, no problem.

**Pull request merged**
